Sources/SourceControl/Repository.swift

```diff
@@ -42,7 +42,9 @@ public struct RepositorySpecifier: Hashable {
 
     /// Returns the cleaned basename for the specifier.
     public var basename: String {
-        var basename = self.location.description.components(separatedBy: "/").last(where: { !$0.isEmpty }) ?? ""
+        guard var basename = self.url.pathComponents.last(where: { !$0.isEmpty }) else {
+            fatalError("URL should have at least one non-empty component")
+        }
```

Comment:
this ie effectively the same as force unwrap (with a better error message). the issue with this approach is that it can cause crashes for processes hosting libSwiftPM like IDEs. given that this did not crash before, I worry about changing this behavior now. a more subtle change would be to default to empty string (as before) but assert on it, which only takes effect in debug builds and as such will not introduce new crashing behavior

Reply:
Ah, I see.